VOKMockUrlProtocol.m

```diff
@@ -124,24 +135,25 @@ - (NSArray *)resourceNames
                 [resourceName appendFormat:AppendSeparatorFormat,
                  [[[NSString alloc] initWithData:bencoded encoding:NSUTF8StringEncoding]
                   stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
-                resourceNames = @[ resourceName, hashResourceName, ];
             } else {
                 // If the bencode fails (or the JSON-decode fails), just use the hash resource name.
-                resourceNames = @[ hashResourceName, ];
+                [resourceNames removeObject:resourceName];
             }
             
         } else {
             // Otherwise, just use the hash resource name.
-            resourceNames = @[ hashResourceName, ];
+            [resourceNames removeObject:resourceName];
         }
         
-    } else {
-        // Otherwise, the only possible name is what we have so far.
-        resourceNames = @[ resourceName, ];
     }
     
-    for (NSMutableString *name in resourceNames) {
+    for (NSMutableString *name in [resourceNames copy]) {
         [self replacePathSeparatorsInMutableString:name];
```

Comment:
This could be swapped to be below the `removeObject:`, which could be made to `continue`... but I do have a tendency to prematurely optimize.